client/my-sites/site/index.jsx

```diff
@@ -16,7 +16,24 @@ module.exports = React.createClass( {
 	displayName: 'Site',
 
 	componentDidMount: function() {
-		debug( 'The Site component is mounted.' );
+		if ( this.props.isSelected ) {
+			this.scrollIntoView();
```

Comment:
I tried that out.  The code is actually a lot uglier that way because you have to assign a `ref` to only the selected site, and the parent container of all the sites as well.  Here's a patch against the current version of this PR:  https://gist.github.com/nylen/01f9adf6d457ccfd5dc7

Also, it doesn't work in the sidebar.  The reason is related to my earlier comment:

> Currently we have both this.props.sites.selected and this.props.selected, we should probably get rid of one of these for clarity.

The site-selector in use in the sidebar doesn't get a `selected` prop and must use `sites.selected` instead.  This makes it impossible to compare against the previous value.

So the `selected` vs `sites.selected` thing needs to be cleaned up first, to use `props.selected` and never `props.sites.selected`.

---

Let's add that prop if and when we need it - otherwise it will be just one more thing laying around in the codebase.